cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -1,12 +1,80 @@
+use std::cell::RefCell;
+use std::collections::VecDeque;
+use std::sync::{Arc, Mutex};
+
 use async_trait::async_trait;
-use catalog_api_v1::types::{self as api_types, error as api_error};
-use catalog_api_v1::{Client as APIClient, Error as APIError};
+use catalog_api_v1::types::{self as api_types, error as api_error, ErrorResponse};
+use catalog_api_v1::{Client as APIClient, Error as APIError, ResponseValue};
 use enum_dispatch::enum_dispatch;
+use reqwest::header::HeaderMap;
+use reqwest::StatusCode;
+use serde::{Deserialize, Serialize};
 use thiserror::Error;
 
 use crate::data::System;
+use crate::models::search::SearchResults;
 
 pub const DEFAULT_CATALOG_URL: &str = "https://flox-catalog.flox.dev";
+pub const FLOX_CATALOG_MOCK_DATA_VAR: &str = "_FLOX_USE_CATALOG_MOCK";
+
+type ResolvedGroups = Vec<ResolvedPackageGroup>;
+
+// Arc allows you to push things into the client from outside the client if necessary
+// Mutex allows you to share across threads (necessary because of tokio)
+// RefCell lets us mutate the field without needing to make the Client trait methods mutable
+type MockField<T> = Arc<Mutex<RefCell<T>>>;
```

Comment:
suggestion: Add comments on how this is used and what "allows you to push things into the client from outside the client" means in this case.
I also have issues figuring out what the scope of a mockfield is supposed to be? 
`read_mock_responses` returns mockfields as a queue/stack of responses, but the type signature represents practically anything?

quesetion/suggestion: Why do we need a `RefCell` here?
My understanding is that `RefCell` allows you get mutable access from a reference _on the same thread_ i.e. a non thread safe `Mutex`. given that we already use a Mutex that gives us metuable references if we need to, adding a refcell to the mix seems a bit redundant 🤔